### PR TITLE
feat(molecule/accordion): add a way to know which tab has been toggled

### PR DESCRIPTION
**Pull request by jordevo**
### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Only
- [x] Refactor

### Description, Motivation and Context
Currently, our `molecule/accordion` component doesn't give us a way to know which tab has been toggled.
This provides a consumer of this component with an `onToggleTab` prop that allows for a callback to know which tab has been toggled via its index. A test has been added.

### Screenshots - Animations
N/A

### CODE REVIEW

File: components/molecule/accordion/src/index.js [hunk 1]
```diff
@@ -4,23 +4,27 @@ import Tab from './Tab'
 
 const BASE_CLASS = 'sui-MoleculeAccordion'
 
-function MoleculeAccordion(props) {
-  const {children, defaultOpenedTabs = [], ...tabProps} = props
-
+const MoleculeAccordion = ({
+  children,
+  defaultOpenedTabs = [],
+  onToggleTab,
```
Comment (jordevo) [inline components/molecule/accordion/src/index.js]:
```suggestion
  onToggleTab = () => {},
```

File: components/molecule/accordion/src/index.js [hunk 2]
```diff
@@ -4,23 +4,27 @@ import Tab from './Tab'
...
   const initialOpenTabs = children.map((_, index) =>
     defaultOpenedTabs.includes(index)
   )
 
   const [openTabs, setOpenTabs] = useState(initialOpenTabs)
 
-  function onToggle(id) {
-    const {withAutoClose} = props
+  const _handleOnToggle = index => {
+    onToggleTab(index)
```
Comment (andresin87) [inline components/molecule/accordion/src/index.js]: The handler must return the Syntetic click event followed by an object with all the interesting elements to be returned

Comment (andresin87) [reply components/molecule/accordion/src/index.js]: It might be triggered at the same lifecycle of other handlers or actions (L29-30)

Comment (andresin87) [reply components/molecule/accordion/src/index.js]: Apply the HoF patterns for the handlers
```js
// L20
const handleOnToggle = index => event => {
	let newOpenTabs = []
	if (withAutoClose) {
		newOpenTabs[index] = openTabs[index] ? undefined : true
	} else {
		newOpenTabs = [...openTabs]
		newOpenTabs[index] = newOpenTabs[index] ? undefined : true
	}
	onToggleTab(event, {index, oppenedTabs: newOpenTabs})
	setOpenTabs([...newOpenTabs])
}
```

```js
// L38
onToggle={handleOnToggle(index)}
```

**Review by quinwacca — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Review by nucliweb — APPROVED** [review]

**Pull request merged**